### DISCUSSION TITLE
Cannot use layout antd with pnpm

### DISCUSSION THREAD

**Discussion by ngocxuan123** [refinedev/refine#1790] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/26795245/167760217-25ad594d-2b12-4f41-b776-a22d260012b5.png)
I'am try use refine antd with single-spa app

**Comment by omeraplak** [comment, marked as answer]
Hey @ngocxuan123 👋,
Thanks for contacting us. Can you make sure all your refine dependencies are the latest version? Also It would be great if you could also share your package.json file. Thank you for your understanding

**Reply by ngocxuan123** [reply]
_Thank you for your reply. this is my package_
**Root-config package.json**
{
  "scripts": {
    "start": "webpack serve --port 9000 --env isLocal",
    "lint": "eslint src",
    "test": "jest --passWithNoTests",
    "format": "prettier --write .",
    "build": "webpack --mode=production",
    "analyze": "webpack --env analyze",
    "prepare": "husky install"
  },
  "devDependencies": {
    "@babel/core": "^7.15.0",
    "@babel/preset-env": "^7.15.0",
    "@types/systemjs": "^6.1.1",
    "babel-loader": "^8.1.0",
    "eslint": "^7.32.0",
    "eslint-config-important-stuff": "^1.1.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^3.4.1",
    "html-webpack-plugin": "^5.3.2",
    "husky": "^7.0.0",
    "jest": "^27.0.6",
    "jest-cli": "^27.0.6",
    "prettier": "^2.3.2",
    "pretty-quick": "^3.1.1",
    "serve": "^12.0.0",
    "webpack": "^5.51.1",
    "webpack-cli": "^4.8.0",
    "webpack-config-single-spa": "^5.0.0",
    "webpack-dev-server": "^4.0.0",
    "webpack-merge": "^5.8.0"
  },
  "dependencies": {
    "single-spa": "^5.9.3",
    "single-spa-layout": "^1.6.0"
  }
}
**Application with refine**
{
  "scripts": {
    "start": "webpack serve",
    "start:standalone": "webpack serve --env standalone",
    "build": "webpack --mode=production",
    "analyze": "webpack --mode=production --env analyze",
    "lint": "eslint src",
    "test": "jest --passWithNoTests",
    "watch-tests": "jest --watch",
    "format": "prettier --write .",
    "coverage": "jest --coverage"
  },
  "husky": {
    "hooks": {
      "pre-commit": "pretty-quick --staged && concurrently yarn:test yarn:lint"
    }
  },
  "devDependencies": {
    "@babel/core": "^7.12.10",
    "@babel/plugin-proposal-class-properties": "^7.10.1",
    "@babel/plugin-proposal-decorators": "^7.12.12",
    "@babel/preset-env": "^7.12.11",
    "@babel/preset-react": "^7.12.10",
    "@testing-library/react": "^11.2.3",
    "@types/jest": "^26.0.20",
    "autoprefixer": "^10.2.1",
    "babel-eslint": "^11.0.0-beta.2",
    "babel-jest": "^26.1.0",
    "babel-loader": "^8.1.0",
    "concurrently": "^5.2.0",
    "eslint": "^7.18.0",
    "eslint-config-prettier": "^7.1.0",
    "eslint-config-react-important-stuff": "^3.0.0",
    "eslint-plugin-prettier": "^3.3.1",
    "husky": "^4.3.8",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^26.1.0",
    "jest-cli": "^26.1.0",
    "prettier": "^2.0.5",
    "pretty-quick": "^3.1.0",
    "query-string": "^6.13.8",
    "react-router": "^5.2.0",
    "react-router-dom": "^5.2.0",
    "rxjs": "^6.5.5",
    "single-spa-react": "^3.0.0",
    "standalone-single-spa-webpack-plugin": "^1.2.2",
    "webpack": "^5.15.0",
    "webpack-cli": "^4.3.1",
    "webpack-config-single-spa-react": "^2.0.0",
    "webpack-dev-server": "^4.0.0-beta.0",
    "webpack-merge": "^5.7.3"
  },
  "dependencies": {
    "@pankod/refine-antd": "^3.16.4",
    "@pankod/refine-core": "^3.16.4",
    "@pankod/refine-react-router-v6": "^3.17.0",
    "@pankod/refine-simple-rest": "^3.16.4",
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
  }
}


===================
single spa required build **librarytarget: system**. Is this the reason?

**Reply by omeraplak** [reply]
Hey @ngocxuan123 ,
According to your dependencies, version 5 of "react-router-dom" and "react-router" is installed. You cannot use "@pankod/refine-react-router-v6" at the same time. Can you delete your "react-router-dom" and "react-router" dependency or upgrade to version 6?

**Reply by ngocxuan123** [reply]
i try update package.json by remove version 5 of "react-router-dom" and "react-router" but can's resolve this.
![image](https://user-images.githubusercontent.com/26795245/167802573-3b6c996c-bea7-414b-a98c-e6bbb3120bbc.png)
i can use without **Layout of refine-antd**. but when use **useTable()** of refine antd this no request data.

**Reply by omeraplak** [reply]
did you delete the `package-lock` and `node_modules` after doing this?

**Reply by ngocxuan123** [reply]
please check in my repository
https://github.com/ngocxuan123/single-spa-refine

**Reply by ngocxuan123** [reply]
i can **useTable** of refine-core but can't use **useTable** refine-antd.
I try to view your lib code but i can't identify the proplem. Please help me PRO

**Reply by ngocxuan123** [reply]
I try to debug and identify the problem as follows:
when i use **useMenu** in **refine-antd**:
![image](https://user-images.githubusercontent.com/26795245/167987346-1a9e1705-dbc5-491f-af0e-8d1f471176e6.png)
but when **clone useMenu** then it sucesss.

**Reply by omeraplak** [reply]
Hey @ngocxuan123 ,
I clone your project but what steps should I follow to run the project?

**Reply by ngocxuan123** [reply]
please go to packages: people, root-config -> run pnpm install.
go to root repo: pnpm start.
required **lerna**

**Reply by omeraplak** [reply]
Thanks! We found the bug, we plan to release a fix this week 🚀

**Reply by omeraplak** [reply]
btw, this issue is not spa related. It's about `pnpm`. If you have a chance to use npm or yarn you can use it.

and Using "MemoryRouter" instead of the Router in the micro-frontend project would be best.
```
import routerProvider, { MemoryRouterComponent } from "@pankod/refine-react-router-v6";

<Refine
      routerProvider={{
        ...routerProvider,
        RouterComponent: MemoryRouterComponent,
      }}
      Layout={Layout}
      resources={[{ name: "posts" }]}
      dataProvider={dataProvider}
    />
```

**Reply by omeraplak** [reply]
Hey @ngocxuan123 ,
We've released `3.19.0` 🚀  Please upgrade all your refine dependencies to the latest version 🎉

btw, please don't forget to use `Memory Router` as I mentioned in my comment above

**Reply by ngocxuan123** [reply]
thank you so much PRO
